src/Riok.Mapperly/Descriptors/MappingBodyBuilders/ObjectMemberMappingBodyBuilder.cs

```diff
@@ -52,50 +66,82 @@ public static void BuildMappingBody(IMembersContainerBuilderContext<IMemberAssig
                 )
             )
             {
-                BuildMemberAssignmentMapping(ctx, sourceMemberPath, new MemberPath(new[] { targetMember }));
+                var targetMemberPath = new MemberPath(new[] { targetMember });
+                targetMembersAdded.Add(targetMemberPath);
+                BuildMemberAssignmentMapping(ctx, sourceMemberPath, targetMemberPath);
                 continue;
             }
 
             if (targetMember.CanSet)
             {
-                ctx.BuilderContext.ReportDiagnostic(
-                    DiagnosticDescriptors.SourceMemberNotFound,
-                    targetMember.Name,
+                sourceMemberNotFoundDiagnosticTargetMembers.Add(targetMember);
+            }
+        }
+
+        foreach (var sourceMember in ctx.SourceMembers.Values)
+        {
+            if (
+                ctx.BuilderContext.SymbolAccessor.TryFindMemberPath(
                     ctx.Mapping.TargetType,
-                    ctx.Mapping.SourceType
-                );
+                    MemberPathCandidateBuilder.BuildMemberPathCandidates(sourceMember.Name),
+                    new ReadOnlyCollection<string>(new List<string>()),
+                    memberNameComparer,
+                    out var targetMemberPath
+                )
+                && !targetMembersAdded.Contains(targetMemberPath)
+                && targetMembersSet.Contains(targetMemberPath.Path[0])
+            )
+            {
+                var sourceMemberPath = new MemberPath(new[] { sourceMember });
+                BuildMemberAssignmentMapping(ctx, sourceMemberPath, targetMemberPath);
+
+                sourceMemberNotFoundDiagnosticTargetMembers.Remove(targetMemberPath.Path[0]);
             }
         }
 
+        foreach (var targetMember in sourceMemberNotFoundDiagnosticTargetMembers)
+        {
+            ctx.BuilderContext.ReportDiagnostic(
+                DiagnosticDescriptors.SourceMemberNotFound,
+                targetMember.Name,
+                ctx.Mapping.TargetType,
+                ctx.Mapping.SourceType
+            );
+        }
+
         ctx.AddDiagnostics();
     }
 
-    private static void BuildMemberAssignmentMapping(
+    private static bool GetMemberPaths(
```

Comment:
Nit: I find a name like `FindConfiguredMemberPaths` or `ResolveConfiguredMemberPaths` more appropriate as this is not a simple get operation but tries to find the actual path.

Reply:
Agree. This is an expensive operation, and naming should reflect that.